test/components/tableTests.ts

```diff
@@ -294,77 +118,346 @@ describe("Tables", () => {
+      const bigFixedComponent01 = new Mocks.FixedSizeComponent(BIG_COMPONENT_SIZE, BIG_COMPONENT_SIZE);
+      const bigFixedComponent10 = new Mocks.FixedSizeComponent(BIG_COMPONENT_SIZE, BIG_COMPONENT_SIZE);
+      const smallFixedComponent = new Mocks.FixedSizeComponent(SMALL_COMPONENT_SIZE, SMALL_COMPONENT_SIZE);
+
+      const table = new Plottable.Components.Table([
+        [unfixedComponent, bigFixedComponent01],
+        [bigFixedComponent10, smallFixedComponent]
+      ]);
 
-      assert.isNull(table.componentAt(0, 1), "returns null if an empty cell is queried");
-      assert.isNull(table.componentAt(-1, 0), "returns null if a negative row index is passed in");
-      assert.isNull(table.componentAt(0, -1), "returns null if a negative column index is passed in");
-      assert.isNull(table.componentAt(9001, 0), "returns null if a row index larger than the number of rows is passed in");
-      assert.isNull(table.componentAt(0, 9001), "returns null if a column index larger than the number of columns is passed in");
+      const expectedMinWidth = 2 * BIG_COMPONENT_SIZE;
+      const expectedMinHeight = 2 * BIG_COMPONENT_SIZE;
+
+      const constrainedSpaceRequest = table.requestedSpace(0, 0);
+      TestMethods.verifySpaceRequest(constrainedSpaceRequest, expectedMinWidth, expectedMinHeight,
+        "requests enough space for all fixed-size Components when space is constrained");
+
+      const exactlyEnoughSpaceRequest = table.requestedSpace(expectedMinWidth, expectedMinHeight);
+      TestMethods.verifySpaceRequest(exactlyEnoughSpaceRequest, expectedMinWidth, expectedMinHeight,
+        "requests enough space for all fixed-size Components when given exactly enough space");
+
+      const extraSpaceRequest = table.requestedSpace(2 * expectedMinWidth, 2 * expectedMinHeight);
+      TestMethods.verifySpaceRequest(extraSpaceRequest, expectedMinWidth, expectedMinHeight,
+        "requests enough space for all fixed-size Components when given extra space");
+    });
+  });
+
+  describe("layout of constituent Components", () => {
+    const SVG_WIDTH = 300;
+    const SVG_HEIGHT = 300;
+
+    function verifyOrigins(rows: Plottable.Component[][], rowPadding = 0, columnPadding = 0) {
+      let expectedOrigin = {
+        x: 0,
+        y: 0
+      };
+      rows.forEach((row, r) => {
+        let maxHeight = 0;
+        expectedOrigin.x = 0;
+        row.forEach((component, c) => {
+          TestMethods.assertPointsClose(component.origin(), expectedOrigin, 0, `Component at [${r}, ${c}] has the correct origin`);
+          expectedOrigin.x += component.width() + columnPadding;
+          maxHeight = maxHeight > component.height() ? maxHeight : component.height();
+        });
+        expectedOrigin.y += maxHeight + rowPadding;
+      });
+    }
+
+    it("divides available width evenly between non-fixed-width Components", () => {
+      const svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
```

Comment:
Would it make more sense to have the width / height be set per test, using some form of multiplier with the number of the rows and columns to prevent rounding errors?  The way it is now, if either `SVG_WIDTH` or if the number of rows change in a test, then the test would break unexpectedly.